### PR TITLE
Add more specific URI strings to Object Explorer

### PR DESCRIPTION
**Pull request by smartguest**
This adds more specific URI to Object Explorer so that sessions can be differentiated between connections that share the same basic connection details but have different properties.

Part of a URI overhaul

### CODE REVIEW

**Review by kburtram — COMMENTED** [review]
@smartguest please don't merge this PR until after we fork for the March release end-of-week.  Thanks!

**Comment by cheenamalhotra** [comment]
@smartguest 

Can you also validate impact on vscode-mssql extension with this PR and update the same to support connection URI similarly?

**Comment by smartguest** [comment]
@cheenamalhotra Will do, although structurally, vscode-mssql is a bit different from ADS and it does not appear at first glance that it uses single string connection IDs (or option keys) on its side, so it will take some time to implement changes to support the new URI

**Comment by cheenamalhotra** [comment]
For now I'd say make sure nothing breaks, so this change should not negatively affect vscode-mssql connections.. (I haven't tested yet) and you can add a new issue/feature request in vscode-mssql to support multiple connections against same target if needed.

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/ObjectExplorerServiceTests.cs
Comment (cheenamalhotra) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/ObjectExplorerServiceTests.cs:59]: Any special reason for this change?

Comment (smartguest) [reply test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/ObjectExplorerServiceTests.cs:59]: The PersistSecurityInfo is an option that is added as a result of taking into every property besides password and connection name (as it's not supported by SQL connections) to the URI.

It is considered a non default option when it is true on the ADS side, and since we want to treat non special options consistently, it will have to be added to the string here.

Comment (smartguest) [reply test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/ObjectExplorerServiceTests.cs:59]: The PersistSecurityInfo is an option that is added as a result of taking into every property besides password and connection name (as it's not supported by SQL connections) to the URI.

It is considered a non default option when it is true on the ADS side, and since we want to treat non special options consistently, it will have to be added to the string here.

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**
